dpe/src/x509.rs

```diff
@@ -895,6 +930,51 @@ impl X509CertWriter<'_> {
         Ok(bytes_written)
     }
 
+    /// Encode ExtendedKeyUsage extension
+    ///
+    /// https://datatracker.ietf.org/doc/html/rfc5280
+    fn encode_extended_key_usage(
+        &mut self,
+        measurements: &MeasurementData,
+    ) -> Result<usize, DpeErrorCode> {
+        let policy_oid = if measurements.is_ca {
+            Self::IDENTITY_LOC_OID
+        } else {
+            Self::ATTEST_LOC_OID
+        };
+
+        // Assumes only one certificate policy is supported.
+        let extended_key_usage_size = Self::get_extended_key_usage_size(measurements, false)?;
+
+        // Encode Extension
+        let mut bytes_written = self.encode_byte(Self::SEQUENCE_TAG)?;
+        bytes_written += self.encode_size_field(extended_key_usage_size)?;
+        bytes_written += self.encode_oid(Self::EXTENDED_KEY_USAGE_OID)?;
+
+        bytes_written += self.encode_byte(Self::BOOL_TAG)?;
+        bytes_written += self.encode_size_field(Self::BOOL_SIZE)?;
+        bytes_written += self.encode_byte(0xFF)?;
+
+        // Extension data is sequence -> octet string. To compute size, wrap
+        // in tagging twice.
+        bytes_written += self.encode_byte(Self::OCTET_STRING_TAG)?;
+        bytes_written += self.encode_size_field(Self::get_structure_size(
+            Self::get_structure_size(policy_oid.len(), /*tagged=*/ true)?,
+            /*tagged=*/ true,
+        )?)?;
+
+        // Sequence size to just a tagged bool
```

Comment:
Not sure I understand this comment

Reply:
I think I copied that from somewhere :) updated the comment